Sources/SwiftSyntaxBuilder/Buildables.swift.gyb

```diff
@@ -0,0 +1,179 @@
+%{
+  from gyb_syntax_support import *
+  from gyb_syntax_support.kinds import lowercase_first_word
+  from gyb_syntax_support.kinds import SYNTAX_BASE_KINDS
+  # -*- mode: Swift -*-
+  # Ignore the following admonition it applies to the resulting .swift file only
+
+  def syntax_buildable_child_type(type_name, syntax_kind, is_token, is_optional=False):
+    if syntax_kind in SYNTAX_BASE_KINDS:
+      buildable_type = syntax_kind + 'Buildable'
+    elif not is_token:
+      buildable_type = syntax_kind
+    else:
+      buildable_type = type_name
+
+    if is_optional:
+      buildable_type += '?'
+
+    return buildable_type
+}%
+//// Automatically Generated From DeclBuildables.swift.gyb.
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+// MARK: Protocols
+
+% for kind in SYNTAX_BASE_KINDS:
+%   if kind != 'SyntaxCollection':
+%     build_kind = 'Syntax' if kind == 'Syntax' else kind + 'Syntax'
+%     if kind == 'Syntax':
+public protocol ${kind}ListBuildable {
+%     else:
+public protocol ${kind}ListBuildable: SyntaxListBuildable {
+%     end
+  func build${kind}List(format: Format, leadingTrivia: Trivia?) -> [${build_kind}]
+}
+
+%     if kind == 'Syntax':
+public protocol ${kind}Buildable: ${kind}ListBuildable {
+%     else:
+public protocol ${kind}Buildable: SyntaxBuildable, ${kind}ListBuildable {
+%     end
+  func build${kind}(format: Format, leadingTrivia: Trivia?) -> ${build_kind}
+}
+
+extension ${kind}Buildable {
+%     if kind != 'Syntax':
+  func build${kind}(format: Format) -> ${build_kind} {
+    build${kind}(format: format, leadingTrivia: nil)
+  }
+%     end
+  public func buildSyntax(format: Format) -> Syntax {
+    buildSyntax(format: format, leadingTrivia: nil)
+  }
+
+  public func buildSyntax(format: Format, leadingTrivia: Trivia?) -> Syntax {
+    Syntax(build${kind}(format: format, leadingTrivia: leadingTrivia))
+  }
+
+  public func build${kind}List(format: Format) -> [${build_kind}] {
+    build${kind}List(format: format, leadingTrivia: nil)
+  }
+
+  public func build${kind}List(format: Format, leadingTrivia: Trivia?) -> [${build_kind}] {
+    [build${kind}(format: format, leadingTrivia: leadingTrivia)]
+  }
+}
+
+%   end
+% end
+
+// MARK: - Buildables
+
+% for node in SYNTAX_NODES:
+%   if node.is_buildable():
+public struct ${node.syntax_kind}: ${node.base_kind}Buildable {
```

Comment:
@ahoppen would it also make sense to add documentation here`
```
%     for line in dedented_lines(node.description):
/// ${line}
%     end
```

Reply:
And maybe also for the `init` and all the children?

Reply:
Yeah, would be good to use the documentation we already have. It doesn’t have to be in this PR though if you’d rather want to get this merged as soon as possible and tackle it in a follow-up PR. However you like it.

Reply:
Will do it another.